lib/src/presentation/widget/home_screen/movie_list_widget.dart

```diff
@@ -15,22 +15,22 @@ class MovieListScreen extends StatefulWidget {
 }
 
 class _MovieListScreenState extends State<MovieListScreen> {
-  MovieRepository movieRepository = MovieRepository();
+  MovieRepositoryImpl movieRepository = MovieRepositoryImpl();
```

Comment:
This should be `IRepository<List<Movie>>`. Depend on abstractions from domain layer, not concretions from data layer